api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/UpdatableClaimantFields.java

```diff
@@ -1,16 +1,107 @@
 package uk.gov.dhsc.htbhf.claimant.model;
 
-import lombok.AllArgsConstructor;
 import lombok.Getter;
+import uk.gov.dhsc.htbhf.claimant.entity.Address;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Enumeration of all the fields on a Claimant that may be updated by the claimant.
  */
-@AllArgsConstructor
-@Getter
 public enum UpdatableClaimantFields {
 
-    EXPECTED_DELIVERY_DATE("expectedDeliveryDate");
+    EXPECTED_DELIVERY_DATE("expectedDeliveryDate"),
+    ADDRESS("address") {
```

Comment:
I added address as it needs special handling. We'd also add other fields such as name etc